src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -67,30 +59,61 @@ public void makeImmutable() {
   }
 
   /**
-   * Return the object from the hash map cast to the right type. Throw an exception if the Object
-   * cannot be cast to that type.
+   * Return the object from the hash map cast to the right type using default Jackson object
+   * deserialization for the passed object type. Throw an exception if the Object cannot be cast to
+   * that type.
    *
    * @param <T> - type of class to expect in the hash map
    * @param key - key to lookup in the hash map
    * @param type - class requested
    * @return null if not found
    * @throws ClassCastException if found, not castable to the requested type
    */
+  @Nullable
   public <T> T get(String key, Class<T> type) {
```

Comment:
Should we `@Deprecated` the existing Object serialization? If we think the current version makes it difficult to introduce changes, and we're adding a new pair of methods, should we encourage clients to migrate to them by marking the existing functions as deprecated? If we don't, what would we want from the new functions to encourage users to use them?
We could add explicit overloads for the primitives we want to support, e.g. `String get(String key)`, `void put(String key, String value)`.

Reply:
I'm on board with this, though the `get` overloads would only differ by return type, so would need to become `getInt()`, `getString()`... 

That said, if we do that I'd prefer to make that a follow-on effort so that we can move the database changes along.

Reply:
I do not think we should deprecate this now, until we have agreed to what we want people to migrate to. I think there is a lot of value in having a default serdes and there are other ways to handle not-upward-compatible changes than writing your own serdes.

One thing I had not considered before, but this interface is returning String. To be open ended on what kind of serdes we support, should we change this to binary? Or is everyone comfortable with a character-based serdes? That would be a more disruptive change to the Stairway database.

Reply:
When we met to discuss this, using binary buffers instead of `String` did come up, but we talked ourselves out of it, mainly due to the database implications.  We did reason that we could support binary formats by UTF-8 encoding with `String`.

And I agree that deprecation is out of scope for this PR.